File: packages/axon-core-ts/src/material.ts
/**
 * Material — anything tracked in PLM/ERP that can flow through production.
 * Categories follow the canonical battery BOM hierarchy used across the suite.
 */
export type MaterialCategory =
  | "cathode-active"
  | "anode-active"
  | "electrolyte"
  | "binder"
  | "conductive-additive"
  | "current-collector-foil"
  | "separator"
  | "tab"
  | "can"
  | "cap"
  | "busbar"
  | "thermal-interface"
  | "adhesive"
  | "fastener"
  | "harness"
  | "sensor"
  | "cell"
  | "module"
  | "pack"
  | "consumable"
  | "other";

export interface Quantity {
  value: number;
  unit: string; // e.g. "kg", "pcs", "L"
}

export interface SpecRange {
  min?: number;
  max?: number;
  /** strict less-than, e.g., "<1 ppm" appears in CoAs */
  lt?: number;
  gt?: number;
  unit: string;
}

export interface MaterialSpec {
  parameters: Record<string, SpecRange>;
}

export interface Material {
  /** Canonical PLM identifier (Teamcenter / D365 item id). */
  id: string;
  name: string;
  category: MaterialCategory;
  spec?: MaterialSpec;
  /** ISO-3166 alpha-2 country of origin. */
  countryOfOrigin?: string;
  /** SDS / safety classification (UN class for li-ion etc.). */
  hazardClass?: string;
}

export interface Supplier {
  id: string;
  name: string;
  approved: boolean;
  rating?: 1 | 2 | 3 | 4 | 5;
}

File: packages/axon-core-py/src/axon_core/material.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel, Field

MaterialCategory = Literal[
    "cathode-active", "anode-active", "electrolyte", "binder",
    "conductive-additive", "current-collector-foil", "separator",
    "tab", "can", "cap", "busbar", "thermal-interface", "adhesive",
    "fastener", "harness", "sensor", "cell", "module", "pack",
    "consumable", "other",
]


class Quantity(BaseModel):
    value: float
    unit: str


class SpecRange(BaseModel):
    min: Optional[float] = None
    max: Optional[float] = None
    lt: Optional[float] = None
    gt: Optional[float] = None
    unit: str


class MaterialSpec(BaseModel):
    parameters: dict[str, SpecRange] = Field(default_factory=dict)


class Material(BaseModel):
    id: str
    name: str
    category: MaterialCategory
    spec: Optional[MaterialSpec] = None
    countryOfOrigin: Optional[str] = None
    hazardClass: Optional[str] = None


class Supplier(BaseModel):
    id: str
    name: str
    approved: bool = False
    rating: Optional[Literal[1, 2, 3, 4, 5]] = None
